bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/api/player/PlayerManager.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.freeboxos.internal.api.player;
+
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.freeboxos.internal.api.ApiHandler;
+import org.openhab.binding.freeboxos.internal.api.ApiVersion;
+import org.openhab.binding.freeboxos.internal.api.FreeboxException;
+import org.openhab.binding.freeboxos.internal.api.ListResponse;
+import org.openhab.binding.freeboxos.internal.api.Response;
+import org.openhab.binding.freeboxos.internal.api.RestManager;
+import org.openhab.binding.freeboxos.internal.api.login.Session.Permission;
+import org.openhab.binding.freeboxos.internal.api.system.DeviceConfig;
+
+/**
+ * The {@link PlayerManager} is the Java class used to handle api requests
+ * related to player
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@NonNullByDefault
+public class PlayerManager extends RestManager {
+    private static String PLAYER_URL = "player";
+
+    public static Permission associatedPermission() {
+        return Permission.PLAYER;
+    }
+
+    private final ApiVersion version;
+
+    public PlayerManager(ApiHandler apiHandler, ApiVersion version) {
+        super(apiHandler, PLAYER_URL);
+        this.version = version;
+    }
+
+    private String buildSubPath(int playerId, String path) {
+        return String.format("%d%s/%s", playerId, version.baseUrl(), path);
+    }
+
+    public List<Player> getPlayers() throws FreeboxException {
+        return getList(PlayersResponse.class, true);
+    }
+
+    public PlayerStatus getPlayerStatus(int id) throws FreeboxException {
+        return get(buildSubPath(id, "status"), PlayerResponse.class, true);
+    }
+
+    public DeviceConfig getConfig(int id) throws FreeboxException {
+        return get(buildSubPath(id, "system"), PlayerConfigurationResponse.class, true);
```

Comment:
It generates this strange path: api/v8/player/<id>/api/v8/system with <id> being my player id.
Even when the player is on, this leads to this API response:
```
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<title>Freebox :: Erreur interne</title>
<link href="/err/err.css" rel="stylesheet" type="text/css" />
</head>

<body>
<div id="info">
    <div id="errorMsg">
      <h3>Erreur interne</h3>
      <p class="desc">Une erreur interne est survenue, réessayez plus tard</p>
    </div>
</div>

</body>
</html>
```

Reply:
Where did you find the documentation for this API ?
I tried with these other paths without no more success:
`api/v8/player/<id>/system`
`api/v8/player/<id>/system/`

Reply:
I found it by deduction and trial/error :) If your server is running it should work.

Reply:
Are you sure of your path? Because it is strange and more important it does not work for me for two different players.

Reply:
Yes, it works on my side. Maybe is this only valid for Delta players ?

Reply:
Strangely it works for my Freebox Revolution Player, not for my Freebox mini 4K.
```
21:17:16.053 [DEBUG] [ng.freeboxos.internal.api.ApiHandler] - executeUrl GET - https://mafreebox.freebox.fr:443/api/v8/player 
21:17:16.183 [TRACE] [ng.freeboxos.internal.api.ApiHandler] - executeUrl GET - https://mafreebox.freebox.fr:443/api/v8/player returned {"success":true,"result":[{"mac":"xxx","stb_type":"stb_v6","id":1,"last_time_reachable":1630523830,"api_available":true,"device_name":"Freebox Player","device_model":"fbx6hd","reachable":true,"uid":"xxx","api_version":"8.1","lan_gids":["ether-xxx"]},{"mac":"yyy","stb_type":"stb_android","id":18,"last_time_reachable":1630523807,"api_available":false,"device_name":"Freebox-Mini-yyy","device_model":"fbx6lc","reachable":true,"lan_gids":["ether-yyy"]}]}
21:17:16.186 [DEBUG] [ng.freeboxos.internal.api.ApiHandler] - executeUrl GET - https://mafreebox.freebox.fr:443/api/v8/player/1/api/v8/system 
21:17:16.255 [TRACE] [ng.freeboxos.internal.api.ApiHandler] - executeUrl GET - https://mafreebox.freebox.fr:443/api/v8/player/1/api/v8/system returned {"success":true,"result":{"model_info":null,"sensors":null,"uptime":"15 minutes 45 secondes","uptime_val":945,"fans":{},"configured":true,"serial":"zzz","firmware_version":"1.3.27.1"}}
21:17:16.261 [DEBUG] [ng.freeboxos.internal.api.ApiHandler] - executeUrl GET - https://mafreebox.freebox.fr:443/api/v8/player/18/api/v8/system 
21:17:16.282 [TRACE] [ng.freeboxos.internal.api.ApiHandler] - executeUrl GET - https://mafreebox.freebox.fr:443/api/v8/player/18/api/v8/system returned <!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<title>Freebox :: Erreur interne</title>
<link href="/err/err.css" rel="stylesheet" type="text/css" />
</head>

<body>
<div id="info">
    <div id="errorMsg">
      <h3>Erreur interne</h3>
      <p class="desc">Une erreur interne est survenue, réessayez plus tard</p>
    </div>
</div>

</body>
</html>

```

Note the field "api_available" which is false for the mini 4K. Mayne this is the trigger to not try calling the second API for this specific player id ?
So we should be able to manage players even when this API is not available.

Also note that for the revolution player, it returns null for model_info. So you can't assume this will be not null. As a consequence, you should take the name (for the thing label) from player.getName() in case it is null.

Reply:
Also note that if the player is OFF, the API fails. This should not make the full discovery job failing. The exception catching is probably not at the right level in this case.

Reply:
In case reachable field is false in the player API response, I think you should even not call the second API. 
And in this case, I believe the thing status should be OFFLINE.

Reply:
I think I managed to address your comments here.
So I do not issue a second api call if api is not available and player not reachable. This should avoid errors at discovery level but still I keep in mind that full configuration of the properties will not have to remain in the discovery but in first thing initialization instead. This will be changed in a second step.